its/src/test/runTest.ts

```diff
@@ -41,32 +41,46 @@ async function main() {
 			stdio: 'inherit'
 		});
 
+		const testErrors = [];
+
 		// run the integration test
-		await runTests({
-			// Use the specified `code` executable
-			vscodeExecutablePath,
-			extensionDevelopmentPath,
-			extensionTestsPath
-		});
+		try {
+			await runTests({
+				// Use the specified `code` executable
+				vscodeExecutablePath,
+				extensionDevelopmentPath,
+				extensionTestsPath
+			});
+		} catch(testError) {
+			testErrors.push(testError);
+		}
 
-        const javaExtensionTestsPath = path.resolve(__dirname, './javaSuite');
+		const javaExtensionTestsPath = path.resolve(__dirname, './javaSuite');
 		const javaTestWorkspace = path.resolve(__dirname, '../../samples/workspace-java.code-workspace');
 
 		cp.spawnSync(cliPath, ['--install-extension', 'redhat.java'], {
 			encoding: 'utf-8',
 			stdio: 'inherit'
 		});
 
-		await runTests({
-			// Use the specified `code` executable
-			vscodeExecutablePath,
-			extensionDevelopmentPath,
-			extensionTestsPath: javaExtensionTestsPath,
-			launchArgs: [javaTestWorkspace]
-		});
+		try {
+			await runTests({
+				// Use the specified `code` executable
+				vscodeExecutablePath,
+				extensionDevelopmentPath,
+				extensionTestsPath: javaExtensionTestsPath,
+				launchArgs: [javaTestWorkspace]
+			});
+		} catch(testError) {
+			testErrors.push(testError);
+		}
+
+		if (testErrors.length > 0) {
+			throw new Error('At least one test suite failed, please check logs above for actual failure.');
```

Comment:
I wonder if you should not simply rethrow `testErrors[0]` to have a better stack trace in case the error is in the test script itself?

Reply:
Actually, I think we can be a bit more clever and provide a nicer developer UX:
- if there is a single error, re-throw it as it is
- if there are several errors, re-throw one with a message that aggregates the other messages